protocols/kad/src/behaviour.rs

```diff
@@ -990,6 +994,40 @@ where
         id
     }
 
+    pub fn set_explicit_mode(&mut self, mode: Option<Mode>) {
```

Comment:
```suggestion
    /// Either set Kademlia [`Mode`] explicitly via `Some(_)` or enable automatic configuration of the Kademlia [`Mode`] based on the external addresses available via `None`.
    pub fn set_mode(&mut self, mode: Option<Mode>) {
```

- Nit pick, passing `None` enables auto-mode, thus the function name (`xxx_explicit_xxx`) is not accurate. Would you agree?
- Adding some documentation. Feel free to rephrase. Just a suggestion.

Reply:
> Nit pick, passing None enables auto-mode, thus the function name (xxx_explicit_xxx) is not accurate. Would you agree?

I disagree. If you call `set_explicit_mode` with `None`, you are **not** setting an explicit mode. Seems quite logical to me?